diag_manager/fms_diag_object.F90

```diff
@@ -872,4 +874,48 @@ subroutine dump_diag_obj( filename )
   call mpp_error( FATAL, "You can not use the modern diag manager without compiling with -Duse_yaml")
 #endif
 end subroutine
+
+!> @brief Allocates outmask(second argument) with sizes of first three dimensions of field(first argument).
+!! Initializes the outmask depending on presence/absence of inmask and rmask.
+!! Uses and sets rmask_threshold.
+subroutine fms_diag_init_mask_3d(field, outmask, rmask_threshold, inmask, rmask)
```

Comment:
This is not a fmsDiagObject_type procedure and does not used the type at all.  It should be somewhere else.  Do you want to start a new file (fms_diag_util.F90) for these small procedures, or put them in diag_util, or make this a type bound procedure?